src/overlays/bankid/index.tsx

```diff
@@ -0,0 +1,280 @@
+import React from "react";
+
+import { IEcomContext, IEcomStore } from "../../types";
+import { isMobile } from "../../utils/device";
+
+import BankIdOverlay from "./base";
+import StoreAction from "../../constants/store-action";
+
+interface IBankIdProps extends IEcomContext, IEcomStore {
+    onHideOverlay: () => void;
+    onProceedToNextStep: () => void;
+}
+
+interface IState {
+    useQrCode: boolean;
+    cancellationToken?: NodeJS.Timeout;
+}
+
+class BankId extends React.Component<IBankIdProps, IState> {
+    constructor(props: IBankIdProps) {
+        super(props);
+
+        const useQrCode = !isMobile();
+        this.state = {
+            useQrCode,
+            cancellationToken: null,
+        };
+
+        this.cancel = this.cancel.bind(this);
```

Comment:
Instead of `.bind()`, one can use arrow functions when declaring methods on a class. It gets a lot of boilerplate out of the way.

Reply:
Agreed. The reason I used bind was to stay consistent to the rest of the code base.